core/templates/dev/head/domain/question/QuestionObjectFactory.ts

```diff
@@ -90,26 +90,32 @@ oppia.factory('QuestionObjectFactory', [
         return 'A solution must be specified';
       }
       var answerGroups = this._stateData.interaction.answerGroups;
-      var taggedMisconceptionIds = {};
+      var taggedSkillMisconceptionIds = {};
       var atLeastOneAnswerCorrect = false;
       for (var i = 0; i < answerGroups.length; i++) {
         if (answerGroups[i].outcome.labelledAsCorrect) {
           atLeastOneAnswerCorrect = true;
           continue;
         }
-        if (answerGroups[i].taggedMisconceptionId !== null) {
-          taggedMisconceptionIds[answerGroups[i].taggedMisconceptionId] = true;
+        if (answerGroups[i].taggedSkillMisconceptionId !== null) {
+          taggedSkillMisconceptionIds[
+            answerGroups[i].taggedSkillMisconceptionId] = true;
         }
       }
       if (!atLeastOneAnswerCorrect) {
         return 'At least one answer should be marked correct';
       }
       var pendingMisconceptionNamesToTag = [];
-      for (var i = 0; i < misconceptions.length; i++) {
-        if (!taggedMisconceptionIds[misconceptions[i].getId()]) {
-          pendingMisconceptionNamesToTag.push(misconceptions[i].getName());
+      Object.keys(misconceptionsPerSkill).forEach(function(skillId) {
+        for (var i = 0; i < misconceptionsPerSkill[skillId].length; i++) {
+          var skillMisconceptionId =
+            skillId + '-' + misconceptionsPerSkill[skillId][i].getId();
+          if (!taggedSkillMisconceptionIds[skillMisconceptionId]) {
```

Comment:
`!taggedSkillMisconceptionIds.hasOwnProperty(skillMisconceptionId)`

(Throughout, please use this approach for membership checking in JS objects.)

Reply:
Done.